### PR TITLE
Add default value of params in pipelinerun and taskrun description

### PR DESCRIPTION
**Pull request by GwonsooLee**
# Changes
#1072 
Show default values of pipeline and task when a user describes the pipelinerun or taskrun. You can check if it is default value or not by describing the task or pipeline. For pipelinerun and taskrun, only applied value will be shown.

Here is the example pipeline
```
apiVersion: tekton.dev/v1beta1
kind: Pipeline
metadata:
  name: hello-goodbye
spec:

  params:
    - name: param1
      type: string
      default: test1
    - name: param2
      type: string
      default: test2
    - name: param3
      type: string
      default: test3
  tasks:
  - name: hello
    taskRef:
      name: hello
    params:
      - name: param1
        value: "$(params.param1)"
      - name: param2
        value: "$(params.param2)"
  - name: goodbye
    runAfter:
     - hello
    taskRef:
      name: goodbye
```

Previous
```
$ tkn p start hello-goodbye -p param1=real1 -p param2=real2
$ tkn pr desc --last
Name:           hello-goodbye-run-8krcq
Namespace:      tekton-pipelines
Pipeline Ref:   hello-goodbye
Timeout:        1h0m0s
Labels:
 tekton.dev/pipeline=hello-goodbye

🌡️  Status

STARTED          DURATION     STATUS
15 seconds ago   15 seconds   Succeeded

📦 Resources

 No resources

⚓ Params

 NAME       VALUE
 ∙ param1   real1
 ∙ param2   real2

🗂  Taskruns

 NAME                                      TASK NAME   STARTED          DURATION    STATUS
 ∙ hello-goodbye-run-8krcq-goodbye-h8x8l   goodbye     9 seconds ago    8 seconds   Succeeded
 ∙ hello-goodbye-run-8krcq-hello-5w6v4     hello       15 seconds ago   6 seconds   Succeeded
```

After 
```
$ tkn pr desc --last
Name:           hello-goodbye-run-8krcq
Namespace:      tekton-pipelines
Pipeline Ref:   hello-goodbye
Timeout:        1h0m0s
Labels:
 tekton.dev/pipeline=hello-goodbye

🌡️  Status

STARTED        DURATION     STATUS
1 minute ago   15 seconds   Succeeded

📦 Resources

 No resources

⚓ Params

 NAME       VALUE
 ∙ param1   real1
 ∙ param2   real2
 ∙ param3   test3

📝 Results

 No results

🗂  Taskruns

 NAME                                      TASK NAME   STARTED        DURATION    STATUS
 ∙ hello-goodbye-run-8krcq-goodbye-h8x8l   goodbye     1 minute ago   8 seconds   Succeeded
 ∙ hello-goodbye-run-8krcq-hello-5w6v4     hello       1 minute ago   6 seconds   Succeeded
```

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes
```release-note
When a user describe pipelinerun or taskrun, all applied values of parameter will be shown.
 In other words, unless a value of the param is passed, default value of it will be shown.
```

### CODE REVIEW

**Comment by GwonsooLee** [comment]
I ran `go generated` again and again but I got an error like this.
```
--- FAIL: TestPipelineRunDescribe_only_taskrun (0.00s)
    describe_test.go:152: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_multiple_taskrun_ordering (0.00s)
    describe_test.go:238: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_failed (0.00s)
    describe_test.go:313: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_failed_withoutTRCondition (0.00s)
    describe_test.go:412: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_failed_withoutPRCondition (0.00s)
    describe_test.go:476: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_with_resources_taskrun (0.00s)
    describe_test.go:552: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_without_start_time (0.00s)
    describe_test.go:594: Unexpected error: no matches for tekton.dev/, Resource=pipelines
--- FAIL: TestPipelineRunDescribe_without_pipelineref (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x20299ea]
```

I think it seems that resources in tb(github.com/tektoncd/pipeline/test/builder) are deprecated. Can you please help me what the problem is..?

**Comment by GwonsooLee** [comment]
Working on it..! :)

**Comment by pradeepitm12** [comment]
@GwonsooLee Thanks for the PR
maybe the error is because we have missed adding pipeline in the resource
e.g. - `cs.Pipeline.Resources = cb.APIResourceList(version, []string{"pipelinerun","pipeline", "taskrun"})`

**Comment by GwonsooLee** [comment]
> @GwonsooLee Thanks for the PR
> maybe the error is because we have missed adding pipeline in the resource
> e.g. - `cs.Pipeline.Resources = cb.APIResourceList(version, []string{"pipelinerun","pipeline", "taskrun"})`

@pradeepitm12 Thank you so much for debugging! 
This is my first PR so that I am not sure my environment is setup well for test. I am now using docker-desktop (k8s version 1.16)

And I run `go mod vender` and `go test -v ./...` but it keeps failing.. 
```
--- FAIL: TestTaskRunDescribe_empty_taskrun (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x202853e]
```
I am working on this now.. but if I am doing something wrong then please let me know.. If I setup well this time then I think I could do more contribution :)

**Comment by pradeepitm12** [comment]
> > @GwonsooLee Thanks for the PR
> > maybe the error is because we have missed adding pipeline in the resource
> > e.g. - `cs.Pipeline.Resources = cb.APIResourceList(version, []string{"pipelinerun","pipeline", "taskrun"})`
> 
> @pradeepitm12 Thank you so much for debugging!
> This is my first PR so that I am not sure my environment is setup well for test. I am now using docker-desktop (k8s version 1.16)
> 
> And I run `go mod vender` and `go test -v ./...` but it keeps failing..
> 
> ```
> --- FAIL: TestTaskRunDescribe_empty_taskrun (0.00s)
> panic: runtime error: invalid memory address or nil pointer dereference [recovered]
>         panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x202853e]
> ```
> 
> I am working on this now.. but if I am doing something wrong then please let me know.. If I setup well this time then I think I could do more contribution :)
Happy to help,
to make things easier you can use the `make check` and `make generated`
for more on environment/setup you can check [here](https://github.com/tektoncd/cli/blob/master/DEVELOPMENT.md)

**Comment by GwonsooLee** [comment]
> cs.Pipeline.Resources

I added `pipeline` to `TestPipelineRunDescribe_only_taskrun` function, and run `make generated` for test, but it failed. Is this because I used pipeline.Get() function from description? (If I removed it, then it works which, of course, it should be if I removed it..)

--- FAIL: TestPipelineRunDescribe_only_taskrun (0.00s)
    describe_test.go:154: Unexpected error: pipelines.tekton.dev "pipeline" not found

**Comment by GwonsooLee** [comment]
Hum.. Not easy to use pipeline in pipelinerun code.. :( 

```
TestPipelinesE2E/Get_list_of_Pipelines_from_namespace__testcli-8jsxq: pipeline_test.go:83: validating Pipelines List command TestPipelinesE2E/Get_list_of_Pipelines_from_namespace__testcli-8jsxq: pipeline_test.go:89: assertion failed: Command: /home/prow/go/src/github.com/tektoncd/cli/tkn pipelines list --namespace testcli-8jsxq ExitCode: 0 Stdout: NAME AGE LAST RUN STARTED DURATION STATUS output-pipeline 2 seconds ago --- --- --- --- Stderr: Failures: Expected stdout to contain "NAME AGE LAST RUN STARTED DURATION STATUS\noutput-pipeline 3 seconds ago --- --- --- ---\n"
```

File: pkg/pipelinerun/description/description.go [hunk 1]
```diff
@@ -152,6 +154,19 @@ func PrintPipelineRunDescription(s *cli.Stream, prName string, p cli.Params) err
...
+		//if err != nil {
+		//	return err
+		//}
+
+		if pl != nil && len(pl.Spec.Params) > 0 {
+			if err := description.SetDefault(&pr.Spec.Params, pl.Spec.Params); err != nil {
+				return err
+			}
+		}
+	}
```
Comment (GwonsooLee) [inline pkg/pipelinerun/description/description.go:168]: @pradeepitm12  I think this part should be reviewed more in strict way because I remove error part.

The reason for this is that if I get `err` and return err if there is, then `go test` failed because pipleline resource does not exist. ( I dived deep this problem but I could solve this yet..) 


```
pl, _ := pipeline.Get(cs, pr.Spec.PipelineRef.Name, metav1.GetOptions{}, p.Namespace())
```
So, I skipped err and if there is an error, it will skip this part (this only occurs in the test environment as far as I know). If there is no error, then SetDefault() function will be called. 

I would like to ask which is the most suitable way to make this part for `pipelinerun`

I've tested as many cases as possible, but with the real manifest file, this works well
